src/StaticWebAssetsSdk/Tasks/ComputeEndpointsForReferenceStaticWebAssets.cs

```diff
@@ -21,12 +21,12 @@ public class ComputeEndpointsForReferenceStaticWebAssets : Task
     public override bool Execute()
     {
         var assets = StaticWebAsset.ToAssetDictionary(Assets);
-        var candidateEndpoints = StaticWebAssetEndpoint.FromItemGroup(CandidateEndpoints);
 
-        var endpoints = new List<StaticWebAssetEndpoint>();
+        var result = CandidateEndpoints;
```

Comment:
Removing the list allocation here and instead reusing the CandidateEndpoints array. This takes advantage of the fact that MSBuild will skip over `null` values on returned arrays.

(That's precisely done so that tasks can avoid allocating extra things)